src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -41,28 +41,29 @@ public T get(int index) {
     @Override
     public T set(T value, int index) {
         checkIndex(index, size - 1);
-        Node<T> removeNode = findNode(index);
-        T removeValue = removeNode.item;
-        removeNode.item = value;
-        return removeValue;
+        Node<T> node = findNode(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
         checkIndex(index, size - 1);
-        return unlink(index);
+        Node<T> removeNode = findNode(index);
+        return unlink(removeNode);
     }
 
     @Override
     public boolean remove(T object) {
         Node<T> currentNode = head;
         for (int i = 0; i < size; i++) {
             if (object == currentNode.item || (object != null && object.equals(currentNode.item))) {
-                unlink(i);
+                Node<T> removeNode = findNode(i);
```

Comment:
should we call method findNode() here?
we can pass `currentNode` to the method unlink()

Reply:
I agree, it makes more sense